sdk/go/common/workspace/config.go

```diff
@@ -129,7 +129,16 @@ func createConfigValue(rawValue interface{}) (config.Value, error) {
 }
 
 func envConfigValue(v esc.Value) config.Plaintext {
+	if v.Unknown {
```

Comment:
esc values can be unknown? 
Another point towards just making config be a full on PropertyValue internally if that's the case.

Reply:
> Another point towards just making config be a full on PropertyValue internally if that's the case.

Honestly I have a very slight preference for making config `esc.Value`s internally:
- The value space is more restrictive than `PropertyValue`s (no assets, archives, resource references)
- `esc.Value`s carry additional metadata about where they are defined (especially relevant for config, as config is now a three-layer merge)
- The APIs for working with `esc.Value`s are a bit more modern than those for `PropertyValue`s

That said, the latter two issues are addressable for `PropertyValue`s. Seems a bit like shoehorning too much functionality into that type for my taste, though.

Reply:
Making them esc.Values sounds interesting except that these need to go over the wire to language runtimes/sdks at some point and we already have an encoding for that (PropertyValues). Definitely an approach to explore though.